UnattendResources/Logon.ps1

```diff
@@ -191,6 +195,11 @@ try
     $disableSwap = Get-IniFileValue -Path $configIniPath -Section "sysprep" -Key "disable_swap" -Default $false -AsBoolean
     $goldImage = Get-IniFileValue -Path $configIniPath -Section "DEFAULT" -Key "gold_image" -Default $false -AsBoolean
     $serialPortName = Get-IniFileValue -Path $configIniPath -Section "cloudbase_init" -Key "serial_logging_port"
+    $productKey = Get-IniFileValue -Path $configIniPath -Section "DEFAULT" -Key "product_key"
+
+    if ($productKey -and $goldImage) {
+        Activate-Windows

```

Comment:
check if golden_image_path has been set, and not if the image should be golden image